tests/functional/test_endpoints_v2.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2012-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import os
+
+import pytest
+
+from botocore.endpoints_v2 import EndpointProvider, EndpointResolutionError
+from botocore.loaders import Loader
+
+_LOADER = Loader()
+
+
+@pytest.fixture(scope="module")
+def partitions():
+    return _LOADER.load_data("partitions")
+
+
+def ruleset_testcases():
+    error_cases = []
+    endpoint_cases = []
+    services = _LOADER.list_available_services("endpoint-rule-set")
+    test_data_dir = os.path.join(
+        os.path.dirname(__file__), "data", "endpoints"
```

Comment:
The test directory should be `tests/functional/endpoint-rules`. Sorry for not telling you sooner.

Reply:
better late than never!